tensorflow_addons/activations/gelu_test.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from absl.testing import parameterized
+
+import math
+
+import numpy as np
+import tensorflow as tf
+from tensorflow_addons.activations import gelu
+from tensorflow_addons.utils import test_utils
+
+
+def _ref_gelu(x, approximate=True):
+    x = tf.convert_to_tensor(x)
+    if approximate:
+        pi = tf.cast(math.pi, x.dtype)
+        coeff = tf.cast(0.044715, x.dtype)
+        return 0.5 * x * (
+            1.0 + tf.tanh(tf.sqrt(2.0 / pi) * (x + coeff * tf.pow(x, 3))))
+    else:
+        return 0.5 * x * (
+            1.0 + tf.math.erf(x / tf.cast(tf.sqrt(2.0), x.dtype)))
+
+
+@test_utils.run_all_in_graph_and_eager_modes
+class TestGelu(tf.test.TestCase, parameterized.TestCase):
+    @parameterized.named_parameters(("float16", np.float16),
+                                    ("float32", np.float32),
+                                    ("float64", np.float64))
```

Comment:
What about testing integer and quantization types?
For integer types, I believe that Gelu will simply behave as Relu.

Reply:
I do not really get your point. Like `int32`, do you mean we first cast it to `float` and do computations in `float` and finally cast back to `int32`? If so, it's weird why users don't cast `int32` to `float` first?

Actually, most of activation ops in core TF can support only floating points input. ReLU/ReLU6 is an exception because `cwiseMax`/`cwiseMin` can run in non-floating dtype.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/nn_ops.cc#L1053-L1144

Reply:
BTW, after a rough computing with google calculator, I found there are some gap between ReLU and GeLU with int type. When input=2, approximate version shows the result of 1.95459769409 and non-approximate version shows the one of 1.9544997361. Get deeper into the definition of GeLU:
```python
gelu(x) = x * P(X <= x) = x * normcdf(x)
``` 
When `x=2`, gelu(2) = 2 * normcdf(2) = 2 * 0.9772 != 2.

[Approximate](https://www.google.com/search?ei=GgBdXbSjPIbW0gSaxaPoAQ&q=0.5+*+2.0+*+%281+%2B+tanh%28sqrt%282+%2F+pi%29+*+%282.0+%2B+0.044715+*+2.0%5E3%29%29%29&oq=0.5+*+2.0+*+%281+%2B+tanh%28sqrt%282+%2F+pi%29+*+%282.0+%2B+0.044715+*+2.0%5E3%29%29%29&gs_l=psy-ab.3...1226.1638..2012...0.0..0.63.118.2......0....1..gws-wiz.GZzLvi8SKuQ&ved=0ahUKEwi0mO-WxZPkAhUGq5QKHZriCB0Q4dUDCAo&uact=5)
[non-approximate](https://www.google.com/search?q=0.5+*+2.0+*+(1.0+%2B+erf(2.0+%2F+sqrt(2.0)))&oq=0.5+*+2.0+*+(1.0+%2B+erf(2.0+%2F+sqrt(2.0)))&aqs=chrome..69i57j6.2966j0j7&sourceid=chrome&ie=UTF-8)

Reply:
@mostafaelhoushi `For integer types, I believe that Gelu will simply behave as Relu.` I don't think that this is true. Can you elaborate a bit on this?

Reply:
Thanks @WindQAQ and @AakashKumarNain for your feedback.
I meant if both the input and _output_ are constrained to be integer, then Gelu will behave as Relu. e.g., 

for the example that @WindQAQ mentioned: `gelu(2) ~= 2*0.9772 - 1.9554`
but if the activations are constrained to be input then we need to round the output to the nearest integer... so `round(gelu(2)) = 2 = relu(2)`

However, @WindQAQ mentioned an important point that  "most of activation ops in core TF (and PyTorch) can support only floating points input. ReLU/ReLU6". Hence, I think you may safely ignore this suggestion.

Reply:
Thanks @WindQAQ and @AakashKumarNain for your feedback.
I meant if both the input and _output_ are constrained to be integer, then Gelu will behave as Relu. e.g., 

for the example that @WindQAQ mentioned: `gelu(2) ~= 2*0.9772 - 1.9554`
but if the activations are constrained to be input then we need to round the output to the nearest integer... so `round(gelu(2)) = 2 = relu(2)`

However, @WindQAQ mentioned an important point that  "most of activation ops in core TF (and PyTorch) can support only floating points input. ReLU/ReLU6". Hence, I think you may safely ignore this suggestion.

Reply:
okay! Thanks again for the review :-)